docs/DeveloperGuide.md

```diff
@@ -1,34 +1,965 @@
+- `extractListScope()` - Extracts the list type. Accepts all, complete,  incomplete or available as acceptable scopes. Invoked with `createListCommand()`. 
+
+which would be invoked based on the command word detected in the user input. 
+
+> ℹ️ **Note:** The Parser also contains checks that ensures the parameters passed are appropriate
+
+The methods that check various parameters include
+- `isModuleCodeValid` - Checks that the module code is a valid module code according to the standard for NUS modules. The method uses regex to check for the valid code. There are 2 overloading methods, one for checking a single instance and another for an arraylist to check through a list of module codes. 
+
+### 3.3.2 Command
+
+#### Description
+The `command` component executes the correct command based on what the parser interprets.
+
+#### Design
+The `command` component consists of an abstract class `Command` and 8 subclasses that inherit from it.
+
+The 8 subclasses are:
+* AddCommand
+* CapCommand
+* DeleteCommand
+* DoneCommand
+* ExitCommand
+* ListCommand
+* ProgressCommand
+* UpdateCommand
+
+The correct command is executed once the `Command` object is created by the parser by executing the `execute()` method in the correct subclass.
+The command execution can affect the `Model` (eg. adding a module).
+At the end of each command execution, different methods in the `Ui` will be called to perform certain actions, such as displaying the list of modules to the user.
+
+Below is a Command class diagram.
+
+![archi](images/CommandClassDiagram.png)
+
+<sup>***Figure 3.3.2.1** UML class diagram for Command class*</sup>
+
+### 3.4 Model Component
+
+The `model` component consists of two main packages that define and deal with data storing issues based on the information 
+provided by the user input. The data storing issues are split into two main categories, what data should be included in for a 
+module and a container managing the module objects. The `module` package holds the information which acts as a blueprint for 
+creating and manipulating module objects while the `list` package consists of a class that defines the way the module objects 
+should be managed and stored.
+
+#### 3.4.1 `module` Package
+
+##### Description
+
+The `module` package consists of classes that are used to define the type of data to be stored in a module object and establish 
+a framework to show how other components can make use of the features in module classes.
+
+##### Design
+
+The `module` package consists of classes related to module objects. An abstract class `Module` is created to hold attributes 
+and methods applicable to all class objects. It is then inherited by all other child module classes. A class diagram illustrating 
+the relationship between the interaction of classes under the module package is shown below.
+
+![archi](images/ModuleClassDiagram.png)
```

Comment:
![archi](https://ay2021s2-cs2113t-w09-2.github.io/tp/images/ModuleClassDiagram.png)

Would this diagram be a bit redundant especially on the list of methods presented given it is exactly the same for all classes 🤔 ? It also looks like being out of date as some methods were not included.